cmake/Tests.cmake

```diff
@@ -166,6 +166,12 @@ endforeach(CHECK)
 FILE(GLOB PEXPECTS CONFIGURE_DEPENDS ${CMAKE_SOURCE_DIR}/tests/pexpects/*.py)
 foreach(PEXPECT ${PEXPECTS})
   get_filename_component(PEXPECT ${PEXPECT} NAME)
+  if(DEFINED ASAN)
+    if("${PEXPECT}" STREQUAL "exit_handlers.py")
```

Comment:
Hmm, this was working before. If this is now broken under ASAN, we should fix it. Any idea what changed? Maybe we should bisect.

Reply:
I think it breaks because the main thread delays until all child threads have exited. The python `spawn.wait()` call only gives a process 0.1 secs after SIGHUP to exit. Maybe we can avoid calling `asan_before_exit()` if not exiting normally.

Reply:
Let's get rid of `asan_before_exit()`. ASAN is not telling us anything interesting here; fish does not try to deallocate all of its memory before exit.